启动时报错

### 启动器版本 Launcher Version

3.5.5-1

### 平台 Platform

Linux 6.4.1-arch2-1

### 问题描述 Problem Description

启动时就出现报错。忽略错误后任然可以启动游戏。

### Java

OpenJDK 64-Bit Server VM (mixed mode, sharing), Oracle Corporation

### 游戏版本 Game Version

_No response_

### 对游戏所做的修改 Modifications Made to the Game

_No response_

### 游戏崩溃报告 Game Crash Report

_No response_

### 启动器崩溃报告 Launcher Crash Report

---- Hello Minecraft! Crash Report ----
  Version: 3.5.SNAPSHOT
  Time: 2023-07-10 18:40:26
  Thread: Thread[#34,IO-2,5,main]

  Content: 
    java.lang.RuntimeException: java.lang.NullPointerException: Cannot invoke "sun.net.www.http.HttpClient.getOutputStream()" because "this.http" is null
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1599)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1581)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getHeaderField(HttpURLConnection.java:3250)
	at java.base/java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:538)
	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:308)
	at org.jackhuang.hmcl.util.io.NetworkUtils.resolveConnection(NetworkUtils.java:152)
	at org.jackhuang.hmcl.util.io.NetworkUtils.doGet(NetworkUtils.java:177)
	at org.jackhuang.hmcl.auth.authlibinjector.AuthlibInjectorDownloader.getLatestArtifactInfo(AuthlibInjectorDownloader.java:116)
	at org.jackhuang.hmcl.auth.authlibinjector.AuthlibInjectorDownloader.update(AuthlibInjectorDownloader.java:93)
	at org.jackhuang.hmcl.auth.authlibinjector.AuthlibInjectorDownloader.checkUpdate(AuthlibInjectorDownloader.java:87)
	at org.jackhuang.hmcl.setting.Accounts.lambda$triggerAuthlibInjectorUpdateCheck$0(Accounts.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1623)
Caused by: java.lang.NullPointerException: Cannot invoke "sun.net.www.http.HttpClient.getOutputStream()" because "this.http" is null
	at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1297)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1127)
	at java.base/sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:179)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1657)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1581)
	at java.base/java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:529)
	... 10 more


-- System Details --
  Operating System: Linux 6.4.1-arch2-1
  System Architecture: x86_64
  Java Architecture: amd64
  Java Version: 20.0.1, N/A
  Java VM Version: OpenJDK 64-Bit Server VM (mixed mode, sharing), Oracle Corporation
  JVM Max Memory: 3625975808
  JVM Total Memory: 50331648
  JVM Free Memory: 30783048

### 启动器日志文件 Launcher Log File

[18:40:26] [org.jackhuang.hmcl.util.SelfDependencyPatcher.patch/INFO] Missing JavaFX dependencies, attempting to patch in missing classes
[18:40:26] [org.jackhuang.hmcl.util.SelfDependencyPatcher.loadFromCache/INFO]  - Loading dependencies...
[18:40:26] [org.jackhuang.hmcl.util.SelfDependencyPatcher.patch/INFO]  - Done!
[18:40:26] [org.jackhuang.hmcl.Launcher.main/INFO] *** HMCL 3.5.SNAPSHOT ***
[18:40:26] [org.jackhuang.hmcl.Launcher.main/INFO] Operating System: Linux 6.4.1-arch2-1
[18:40:26] [org.jackhuang.hmcl.Launcher.main/INFO] System Architecture: x86_64
[18:40:26] [org.jackhuang.hmcl.Launcher.main/INFO] Java Architecture: amd64
[18:40:26] [org.jackhuang.hmcl.Launcher.main/INFO] Java Version: 20.0.1, N/A
[18:40:26] [org.jackhuang.hmcl.Launcher.main/INFO] Java VM Version: OpenJDK 64-Bit Server VM (mixed mode, sharing), Oracle Corporation
[18:40:26] [org.jackhuang.hmcl.Launcher.main/INFO] Java Home: /usr/lib/jvm/java-20-openjdk
[18:40:26] [org.jackhuang.hmcl.Launcher.main/INFO] Current Directory: /home/shinonome/.config/hmcl
[18:40:26] [org.jackhuang.hmcl.Launcher.main/INFO] HMCL Directory: /home/shinonome/.local/share/hmcl
[18:40:26] [org.jackhuang.hmcl.Launcher.main/INFO] HMCL Jar Path: /usr/share/java/hmcl/hmcl.jar
[18:40:26] [org.jackhuang.hmcl.Launcher.main/INFO] Memory: 3458MB
[18:40:26] [org.jackhuang.hmcl.Launcher.main/INFO] Metaspace: 5MB
[18:40:26] [org.jackhuang.hmcl.Launcher.main/INFO] XDG Session Type: x11
[18:40:26] [org.jackhuang.hmcl.Launcher.start/INFO] JavaFX Version: 19.0.2.1+1
[18:40:26] [org.jackhuang.hmcl.Launcher.start/INFO] Prism pipeline: com.sun.prism.es2.ES2Pipeline
[18:40:26] [org.jackhuang.hmcl.setting.ConfigHolder.init/INFO] Config location: hmcl.json
[18:40:26] [org.jackhuang.hmcl.auth.authlibinjector.AuthlibInjectorDownloader.checkUpdate/INFO] Checking update of authlib-injector
[18:40:26] [org.jackhuang.hmcl.task.AsyncTaskExecutor.lambda$executeNormalTask$19/FINE] Executing task: org.jackhuang.hmcl.game.GameRepository.refreshVersionsAsync(GameRepository.java:90)
[18:40:26] [org.jackhuang.hmcl.task.AsyncTaskExecutor.lambda$executeNormalTask$19/FINE] Executing task: org.jackhuang.hmcl.game.GameRepository.refreshVersionsAsync(GameRepository.java:90)
[18:40:26] [org.jackhuang.hmcl.event.EventBus.fireEvent/INFO] RefreshingVersionsEvent [source=HMCLGameRepository [versions=null, baseDirectory=/home/shinonome/.minecraft]] gets fired
[18:40:26] [org.jackhuang.hmcl.event.EventBus.fireEvent/INFO] RefreshingVersionsEvent [source=HMCLGameRepository [versions=null, baseDirectory=/home/shinonome/.minecraft]] gets fired
[18:40:26] [org.jackhuang.hmcl.ui.Controllers.initialize/INFO] Start initializing application
[18:40:26] [org.jackhuang.hmcl.event.EventBus.fireEvent/INFO] LoadedOneVersionEvent [source=HMCLGameRepository [versions=null, baseDirectory=/home/shinonome/.minecraft], version=Version [id=FlyingRoad3]] gets fired
[18:40:26] [org.jackhuang.hmcl.event.EventBus.fireEvent/INFO] LoadedOneVersionEvent [source=HMCLGameRepository [versions=null, baseDirectory=/home/shinonome/.minecraft], version=Version [id=FlyingRoad3]] gets fired
[18:40:26] [org.jackhuang.hmcl.event.EventBus.fireEvent/INFO] LoadedOneVersionEvent [source=HMCLGameRepository [versions=null, baseDirectory=/home/shinonome/.minecraft], version=Version [id=诸神匠心1.5]] gets fired
[18:40:26] [org.jackhuang.hmcl.event.EventBus.fireEvent/INFO] LoadedOneVersionEvent [source=HMCLGameRepository [versions=null, baseDirectory=/home/shinonome/.minecraft], version=Version [id=诸神匠心1.5]] gets fired
[18:40:26] [org.jackhuang.hmcl.event.EventBus.fireEvent/INFO] LoadedOneVersionEvent [source=HMCLGameRepository [versions=null, baseDirectory=/home/shinonome/.minecraft], version=Version [id=XPlus 2.0 based on Minecraft 1.19.4 (Fabric)]] gets fired
[18:40:26] [org.jackhuang.hmcl.event.EventBus.fireEvent/INFO] LoadedOneVersionEvent [source=HMCLGameRepository [versions=null, baseDirectory=/home/shinonome/.minecraft], version=Version [id=XPlus 2.0 based on Minecraft 1.19.4 (Fabric)]] gets fired
[18:40:26] [org.jackhuang.hmcl.game.DefaultGameRepository.refreshVersionsImpl/WARNING] Ignoring version fabric-loader-0.14.21-1.20.1 because it inherits from a nonexistent version.
[18:40:26] [org.jackhuang.hmcl.game.DefaultGameRepository.refreshVersionsImpl/WARNING] Ignoring version fabric-loader-0.14.21-1.20.1 because it inherits from a nonexistent version.
[18:40:26] [org.jackhuang.hmcl.event.EventBus.fireEvent/INFO] LoadedOneVersionEvent [source=HMCLGameRepository [versions=null, baseDirectory=/home/shinonome/.minecraft], version=Version [id=XPlus 2.0 based on Minecraft 1.18.2 (Fabric)]] gets fired
[18:40:26] [org.jackhuang.hmcl.event.EventBus.fireEvent/INFO] LoadedOneVersionEvent [source=HMCLGameRepository [versions=null, baseDirectory=/home/shinonome/.minecraft], version=Version [id=XPlus 2.0 based on Minecraft 1.18.2 (Fabric)]] gets fired
[18:40:26] [org.jackhuang.hmcl.event.EventBus.fireEvent/INFO] RefreshedVersionsEvent [source=HMCLGameRepository [versions=[FlyingRoad3, XPlus 2.0 based on Minecraft 1.18.2 (Fabric), XPlus 2.0 based on Minecraft 1.19.4 (Fabric), 诸神匠心1.5], baseDirectory=/home/shinonome/.minecraft]] gets fired
[18:40:26] [org.jackhuang.hmcl.event.EventBus.fireEvent/INFO] RefreshedVersionsEvent [source=HMCLGameRepository [versions=[FlyingRoad3, XPlus 2.0 based on Minecraft 1.18.2 (Fabric), XPlus 2.0 based on Minecraft 1.19.4 (Fabric), 诸神匠心1.5], baseDirectory=/home/shinonome/.minecraft]] gets fired
[18:40:26] [org.jackhuang.hmcl.task.AsyncTaskExecutor.lambda$executeNormalTask$29/FINER] Task finished: org.jackhuang.hmcl.game.GameRepository.refreshVersionsAsync(GameRepository.java:90)
[18:40:26] [org.jackhuang.hmcl.task.AsyncTaskExecutor.lambda$executeNormalTask$29/FINER] Task finished: org.jackhuang.hmcl.game.GameRepository.refreshVersionsAsync(GameRepository.java:90)
[18:40:26] [org.jackhuang.hmcl.util.CrashReporter.uncaughtException/SEVERE] Uncaught exception in thread IO-3
java.lang.NullPointerException: Cannot invoke "sun.net.www.http.HttpClient.getOutputStream()" because "this.http" is null
	at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1297)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1127)
	at java.base/sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:179)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1657)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1581)
	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:224)
	at java.base/java.net.URL.openStream(URL.java:1317)
	at org.jackhuang.hmcl.auth.authlibinjector.AuthlibInjectorServer.refreshMetadata(AuthlibInjectorServer.java:162)
	at org.jackhuang.hmcl.auth.authlibinjector.AuthlibInjectorServer.fetchMetadataResponse(AuthlibInjectorServer.java:156)
	at org.jackhuang.hmcl.setting.Accounts.lambda$init$10(Accounts.java:342)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1623)
[18:40:26] [org.jackhuang.hmcl.util.CrashReporter.uncaughtException/SEVERE] Uncaught exception in thread IO-2
java.lang.RuntimeException: java.lang.NullPointerException: Cannot invoke "sun.net.www.http.HttpClient.getOutputStream()" because "this.http" is null
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1599)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1581)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getHeaderField(HttpURLConnection.java:3250)
	at java.base/java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:538)
	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:308)
	at org.jackhuang.hmcl.util.io.NetworkUtils.resolveConnection(NetworkUtils.java:152)
	at org.jackhuang.hmcl.util.io.NetworkUtils.doGet(NetworkUtils.java:177)
	at org.jackhuang.hmcl.auth.authlibinjector.AuthlibInjectorDownloader.getLatestArtifactInfo(AuthlibInjectorDownloader.java:116)
	at org.jackhuang.hmcl.auth.authlibinjector.AuthlibInjectorDownloader.update(AuthlibInjectorDownloader.java:93)
	at org.jackhuang.hmcl.auth.authlibinjector.AuthlibInjectorDownloader.checkUpdate(AuthlibInjectorDownloader.java:87)
	at org.jackhuang.hmcl.setting.Accounts.lambda$triggerAuthlibInjectorUpdateCheck$0(Accounts.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1623)
Caused by: java.lang.NullPointerException: Cannot invoke "sun.net.www.http.HttpClient.getOutputStream()" because "this.http" is null
	at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1297)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1127)
	at java.base/sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:179)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1657)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1581)
	at java.base/java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:529)
	... 10 more
[18:40:26] [org.jackhuang.hmcl.util.CrashReporter.uncaughtException/SEVERE] ---- Hello Minecraft! Crash Report ----
  Version: 3.5.SNAPSHOT
  Time: 2023-07-10 18:40:26
  Thread: Thread[#35,IO-3,5,main]

  Content: 
    java.lang.NullPointerException: Cannot invoke "sun.net.www.http.HttpClient.getOutputStream()" because "this.http" is null
	at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1297)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1127)
	at java.base/sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:179)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1657)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1581)
	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:224)
	at java.base/java.net.URL.openStream(URL.java:1317)
	at org.jackhuang.hmcl.auth.authlibinjector.AuthlibInjectorServer.refreshMetadata(AuthlibInjectorServer.java:162)
	at org.jackhuang.hmcl.auth.authlibinjector.AuthlibInjectorServer.fetchMetadataResponse(AuthlibInjectorServer.java:156)
	at org.jackhuang.hmcl.setting.Accounts.lambda$init$10(Accounts.java:342)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1623)


-- System Details --
  Operating System: Linux 6.4.1-arch2-1
  System Architecture: x86_64
  Java Architecture: amd64
  Java Version: 20.0.1, N/A
  Java VM Version: OpenJDK 64-Bit Server VM (mixed mode, sharing), Oracle Corporation
  JVM Max Memory: 3625975808
  JVM Total Memory: 50331648
  JVM Free Memory: 30783048

[18:40:26] [org.jackhuang.hmcl.util.CrashReporter.uncaughtException/SEVERE] ---- Hello Minecraft! Crash Report ----
  Version: 3.5.SNAPSHOT
  Time: 2023-07-10 18:40:26
  Thread: Thread[#34,IO-2,5,main]

  Content: 
    java.lang.RuntimeException: java.lang.NullPointerException: Cannot invoke "sun.net.www.http.HttpClient.getOutputStream()" because "this.http" is null
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1599)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1581)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getHeaderField(HttpURLConnection.java:3250)
	at java.base/java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:538)
	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:308)
	at org.jackhuang.hmcl.util.io.NetworkUtils.resolveConnection(NetworkUtils.java:152)
	at org.jackhuang.hmcl.util.io.NetworkUtils.doGet(NetworkUtils.java:177)
	at org.jackhuang.hmcl.auth.authlibinjector.AuthlibInjectorDownloader.getLatestArtifactInfo(AuthlibInjectorDownloader.java:116)
	at org.jackhuang.hmcl.auth.authlibinjector.AuthlibInjectorDownloader.update(AuthlibInjectorDownloader.java:93)
	at org.jackhuang.hmcl.auth.authlibinjector.AuthlibInjectorDownloader.checkUpdate(AuthlibInjectorDownloader.java:87)
	at org.jackhuang.hmcl.setting.Accounts.lambda$triggerAuthlibInjectorUpdateCheck$0(Accounts.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1623)
Caused by: java.lang.NullPointerException: Cannot invoke "sun.net.www.http.HttpClient.getOutputStream()" because "this.http" is null
	at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1297)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1127)
	at java.base/sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:179)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1657)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1581)
	at java.base/java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:529)
	... 10 more


-- System Details --
  Operating System: Linux 6.4.1-arch2-1
  System Architecture: x86_64
  Java Architecture: amd64
  Java Version: 20.0.1, N/A
  Java VM Version: OpenJDK 64-Bit Server VM (mixed mode, sharing), Oracle Corporation
  JVM Max Memory: 3625975808
  JVM Total Memory: 50331648
  JVM Free Memory: 30783048

[18:40:27] [org.jackhuang.hmcl.task.AsyncTaskExecutor.lambda$executeNormalTask$19/FINE] Executing task: org.jackhuang.hmcl.ui.Controllers.initialize(Controllers.java:197)
[18:40:27] [org.jackhuang.hmcl.util.platform.JavaVersion.lambda$lookupJavas$2/FINER] Looking for Java:/usr/lib/jvm/java-17-openjdk/bin/java
[18:40:27] [org.jackhuang.hmcl.util.platform.JavaVersion.lambda$lookupJavas$2/FINER] Looking for Java:/usr/lib/jvm/java-8-openjdk/jre/bin/java
[18:40:27] [org.jackhuang.hmcl.util.platform.JavaVersion.lambda$lookupJavas$2/FINER] Looking for Java:/usr/lib/jvm/java-20-jdk/bin/java
[18:40:27] [org.jackhuang.hmcl.util.platform.JavaVersion.lambda$lookupJavas$2/FINE] Found Java (20.0.1) /usr/lib/jvm/java-20-jdk/bin/java
[18:40:27] [org.jackhuang.hmcl.util.platform.JavaVersion.lambda$lookupJavas$2/FINE] Found Java (1.8.0_372) /usr/lib/jvm/java-8-openjdk/jre/bin/java
[18:40:27] [org.jackhuang.hmcl.util.platform.JavaVersion.lambda$lookupJavas$2/FINE] Found Java (17.0.7) /usr/lib/jvm/java-17-openjdk/bin/java
[18:40:27] [org.jackhuang.hmcl.util.platform.JavaVersion.initialize/FINE] Finished Java installation lookup, found 4
[18:40:27] [org.jackhuang.hmcl.task.AsyncTaskExecutor.lambda$executeNormalTask$29/FINER] Task finished: org.jackhuang.hmcl.ui.Controllers.initialize(Controllers.java:197)
[18:40:27] [org.jackhuang.hmcl.upgrade.IntegrityChecker.isSelfVerified/WARNING] Failed to verify myself, is the JAR corrupt?
java.io.IOException: Signature is missing
	at org.jackhuang.hmcl.upgrade.IntegrityChecker.verifyJar(IntegrityChecker.java:88)
	at org.jackhuang.hmcl.upgrade.IntegrityChecker.requireVerifiedJar(IntegrityChecker.java:106)
	at org.jackhuang.hmcl.upgrade.IntegrityChecker.verifySelf(IntegrityChecker.java:146)
	at org.jackhuang.hmcl.upgrade.IntegrityChecker.isSelfVerified(IntegrityChecker.java:128)
	at org.jackhuang.hmcl.util.CrashReporter.lambda$uncaughtException$0(CrashReporter.java:108)
	at javafx.graphics@19.0.2.1/com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:457)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:400)
	at javafx.graphics@19.0.2.1/com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:456)
	at javafx.graphics@19.0.2.1/com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:96)
	at javafx.graphics@19.0.2.1/com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
	at javafx.graphics@19.0.2.1/com.sun.glass.ui.gtk.GtkApplication.lambda$runLoop$11(GtkApplication.java:316)
	at java.base/java.lang.Thread.run(Thread.java:1623)
[18:40:27] [org.jackhuang.hmcl.upgrade.UpdateChecker.lambda$requestCheckUpdate$2/WARNING] Failed to check for update
java.io.IOException: Self verification failed
	at org.jackhuang.hmcl.upgrade.UpdateChecker.checkUpdate(UpdateChecker.java:85)
	at org.jackhuang.hmcl.upgrade.UpdateChecker.lambda$requestCheckUpdate$2(UpdateChecker.java:109)
	at java.base/java.lang.Thread.run(Thread.java:1623)
[18:42:46] [org.jackhuang.hmcl.task.AsyncTaskExecutor.lambda$executeNormalTask$19/FINE] Executing task: org.jackhuang.hmcl.ui.versions.VersionSettingsPage.initialize(VersionSettingsPage.java:434)
[18:42:46] [org.jackhuang.hmcl.task.AsyncTaskExecutor.lambda$executeNormalTask$19/FINE] Executing task: org.jackhuang.hmcl.ui.versions.VersionSettingsPage.initialize(VersionSettingsPage.java:434)
[18:42:46] [org.jackhuang.hmcl.task.AsyncTaskExecutor.lambda$executeNormalTask$29/FINER] Task finished: org.jackhuang.hmcl.ui.versions.VersionSettingsPage.initialize(VersionSettingsPage.java:434)
[18:42:46] [org.jackhuang.hmcl.ui.construct.Navigator.navigate/INFO] Navigate to LauncherSettingsPage@35b4132c[styleClass=gray-background]
[18:42:46] [org.jackhuang.hmcl.task.AsyncTaskExecutor.lambda$executeNormalTask$29/FINER] Task finished: org.jackhuang.hmcl.ui.versions.VersionSettingsPage.initialize(VersionSettingsPage.java:434)
[18:42:49] [org.jackhuang.hmcl.ui.main.SettingsPage.lambda$onExportLogs$5/INFO] Exporting logs to /home/shinonome/.config/hmcl/hmcl-exported-logs-2023-07-10T18-42-49.log

--- a/HMCLCore/src/main/java/org/jackhuang/hmcl/auth/authlibinjector/AuthlibInjectorDownloader.java
+++ b/HMCLCore/src/main/java/org/jackhuang/hmcl/auth/authlibinjector/AuthlibInjectorDownloader.java
@@ -22,7 +22,6 @@ import com.google.gson.annotations.SerializedName;
 import org.jackhuang.hmcl.download.DownloadProvider;
 import org.jackhuang.hmcl.task.FileDownloadTask;
 import org.jackhuang.hmcl.task.FileDownloadTask.IntegrityCheck;
-import org.jackhuang.hmcl.util.gson.JsonUtils;
 import org.jackhuang.hmcl.util.io.HttpRequest;
 
 import java.io.IOException;
@@ -112,10 +111,7 @@ public class AuthlibInjectorDownloader implements AuthlibInjectorArtifactProvide
 
     private AuthlibInjectorVersionInfo getLatestArtifactInfo() throws IOException {
         try {
-            return JsonUtils.fromNonNullJson(
-                    HttpRequest.GET(downloadProvider.get().injectURL(LATEST_BUILD_URL)).getString(),
-                    AuthlibInjectorVersionInfo.class
-            );
+            return HttpRequest.GET(downloadProvider.get().injectURL(LATEST_BUILD_URL)).getJson(AuthlibInjectorVersionInfo.class);
         } catch (JsonParseException e) {
             throw new IOException("Malformed response", e);
         }

--- a/HMCLCore/src/main/java/org/jackhuang/hmcl/auth/authlibinjector/AuthlibInjectorDownloader.java
+++ b/HMCLCore/src/main/java/org/jackhuang/hmcl/auth/authlibinjector/AuthlibInjectorDownloader.java
@@ -23,7 +23,7 @@ import org.jackhuang.hmcl.download.DownloadProvider;
 import org.jackhuang.hmcl.task.FileDownloadTask;
 import org.jackhuang.hmcl.task.FileDownloadTask.IntegrityCheck;
 import org.jackhuang.hmcl.util.gson.JsonUtils;
-import org.jackhuang.hmcl.util.io.NetworkUtils;
+import org.jackhuang.hmcl.util.io.HttpRequest;
 
 import java.io.IOException;
 import java.net.URL;
@@ -102,7 +102,7 @@ public class AuthlibInjectorDownloader implements AuthlibInjectorArtifactProvide
                     Optional.ofNullable(latest.checksums.get("sha256"))
                             .map(checksum -> new IntegrityCheck("SHA-256", checksum))
                             .orElse(null))
-                                    .run();
+                    .run();
         } catch (Exception e) {
             throw new IOException("Failed to download authlib-injector", e);
         }
@@ -113,9 +113,9 @@ public class AuthlibInjectorDownloader implements AuthlibInjectorArtifactProvide
     private AuthlibInjectorVersionInfo getLatestArtifactInfo() throws IOException {
         try {
             return JsonUtils.fromNonNullJson(
-                    NetworkUtils.doGet(
-                            new URL(downloadProvider.get().injectURL(LATEST_BUILD_URL))),
-                    AuthlibInjectorVersionInfo.class);
+                    HttpRequest.GET(downloadProvider.get().injectURL(LATEST_BUILD_URL)).getString(),
+                    AuthlibInjectorVersionInfo.class
+            );
         } catch (JsonParseException e) {
             throw new IOException("Malformed response", e);
         }

--- a/HMCLCore/src/main/java/org/jackhuang/hmcl/auth/authlibinjector/AuthlibInjectorServer.java
+++ b/HMCLCore/src/main/java/org/jackhuang/hmcl/auth/authlibinjector/AuthlibInjectorServer.java
@@ -17,12 +17,9 @@
  */
 package org.jackhuang.hmcl.auth.authlibinjector;
 
-import static java.nio.charset.StandardCharsets.UTF_8;
 import static java.util.Collections.emptyMap;
 import static org.jackhuang.hmcl.util.Lang.tryCast;
 import static org.jackhuang.hmcl.util.Logging.LOG;
-import static org.jackhuang.hmcl.util.io.IOUtils.readFullyAsByteArray;
-import static org.jackhuang.hmcl.util.io.IOUtils.readFullyWithoutClosing;
 
 import java.io.IOException;
 import java.lang.reflect.Type;
@@ -35,6 +32,8 @@ import java.util.Optional;
 import java.util.logging.Level;
 
 import org.jackhuang.hmcl.auth.yggdrasil.YggdrasilService;
+import org.jackhuang.hmcl.util.io.HttpRequest;
+import org.jackhuang.hmcl.util.io.IOUtils;
 import org.jackhuang.hmcl.util.javafx.ObservableHelper;
 import org.jetbrains.annotations.Nullable;
 
@@ -77,7 +76,7 @@ public class AuthlibInjectorServer implements Observable {
 
             try {
                 AuthlibInjectorServer server = new AuthlibInjectorServer(url);
-                server.refreshMetadata(readFullyWithoutClosing(conn.getInputStream()));
+                server.refreshMetadata(IOUtils.readFullyAsStringWithClosing(conn.getInputStream()));
                 return server;
             } finally {
                 conn.disconnect();
@@ -159,12 +158,11 @@ public class AuthlibInjectorServer implements Observable {
     }
 
     public void refreshMetadata() throws IOException {
-        refreshMetadata(readFullyAsByteArray(new URL(url).openStream()));
+        refreshMetadata(HttpRequest.GET(url).getString());
     }
 
-    private void refreshMetadata(byte[] rawResponse) throws IOException {
+    private void refreshMetadata(String text) throws IOException {
         long timestamp = System.currentTimeMillis();
-        String text = new String(rawResponse, UTF_8);
         try {
             setMetadataResponse(text, timestamp);
         } catch (JsonParseException e) {

--- a/HMCLCore/src/main/java/org/jackhuang/hmcl/util/io/HttpRequest.java
+++ b/HMCLCore/src/main/java/org/jackhuang/hmcl/util/io/HttpRequest.java
@@ -29,7 +29,6 @@ import java.io.OutputStream;
 import java.lang.reflect.Type;
 import java.net.HttpURLConnection;
 import java.net.MalformedURLException;
-import java.net.SocketTimeoutException;
 import java.net.URL;
 import java.util.HashMap;
 import java.util.HashSet;
@@ -218,16 +217,21 @@ public abstract class HttpRequest {
     }
 
     private static String getStringWithRetry(ExceptionalSupplier<String, IOException> supplier, int retryTimes) throws IOException {
-        SocketTimeoutException exception = null;
+        Throwable exception = null;
         for (int i = 0; i < retryTimes; i++) {
             try {
                 return supplier.get();
-            } catch (SocketTimeoutException e) {
+            } catch (Throwable e) {
                 exception = e;
             }
         }
-        if (exception != null)
-            throw exception;
+        if (exception != null) {
+            if (exception instanceof IOException) {
+                throw (IOException) exception;
+            } else {
+                throw new IOException(exception);
+            }
+        }
         throw new IOException("retry 0");
     }
 

--- a/HMCLCore/src/main/java/org/jackhuang/hmcl/util/io/IOUtils.java
+++ b/HMCLCore/src/main/java/org/jackhuang/hmcl/util/io/IOUtils.java
@@ -44,6 +44,12 @@ public final class IOUtils {
         return result.toByteArray();
     }
 
+    public static String readFullyAsStringWithClosing(InputStream stream) throws IOException {
+        ByteArrayOutputStream result = new ByteArrayOutputStream(Math.max(stream.available(), 32));
+        copyTo(stream, result);
+        return result.toString("UTF-8");
+    }
+
     /**
      * Read all bytes to a buffer from given input stream, and close the input stream finally.
      *

